crates/core/src/surfnet/locker.rs

```diff
@@ -311,6 +314,57 @@ impl SurfnetSvmLocker {
         }
         Ok(results.with_new_value(combined))
     }
+
+    /// Retrieves largest accounts from local cache, returning a contextualized result.
+    pub fn get_largest_accounts_local(&self) -> SvmAccessContext<Vec<RpcAccountBalance>> {
+        self.with_contextualized_svm_reader(|svm_reader| {
+            svm_reader
+                .accounts_registry
+                .iter()
+                .sorted_by(|a, b| b.1.lamports.cmp(&a.1.lamports))
+                .take(20)
+                .map(|(pubkey, account)| RpcAccountBalance {
+                    address: pubkey.to_string(),
+                    lamports: account.lamports,
+                })
+                .collect()
+        })
+    }
+
+    pub async fn get_largest_accounts_local_then_remote(
+        &self,
+        client: &SurfnetRemoteClient,
+        config: Option<RpcLargestAccountsConfig>,
```

Comment:
We're passing this config to the remote client, which will filter results by circulating/non-circulating accounts.

However, we aren't using this filter on the local results. The svm has `non_circulating_accounts` on it that can be used for the filter.

This is a bit confusing but I think we should be:
 1. filtering the results of `get_largest_accounts_local`
 2. when we do `remote_client.get_largest_accounts`, we should always _remove_ the filter (so it returns all largest accounts) _then_ we manually apply the filter based on our local svm's `non_circulating_accounts` setting

Reply:
To add some clarity, here is what I think we should do for this method:
 1. Make two requests to the remote - one for the largest circulating accounts, one for the largest non-circulating accounts
 2. If those accounts don't yet exist locally, insert them
 3. Pull from our local cache to find the largest accounts, with the filter applied. If the filter is provided, it should be looking at the `SurfnetSvm`'s `non_circulating_accounts` to see what accounts are circulating/non-circulating